src/app/code/community/Shopgate/Cloudapi/Model/Frontend/Observer/CustomerRegisterSuccess.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright Shopgate Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @author    Shopgate Inc, 804 Congress Ave, Austin, Texas 78701 <interfaces@shopgate.com>
+ * @copyright Shopgate Inc
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ */
+
+class Shopgate_Cloudapi_Model_Frontend_Observer_CustomerRegisterSuccess
+{
+    /**
+     * Checks if the order is received from Shopgate API call.
+     *
+     * @param Varien_Event_Observer $observer
+     */
+    public function execute(Varien_Event_Observer $observer)
+    {
+        /* @todo - Check is shopgate App register */
+        /** @var Mage_Customer_Model_Customer $customer */
+        $customer = $observer->getEvent()->getCustomer();
+
+        /** @var Mage_Customer_AccountController $accountController */
+        $accountController = $observer->getEvent()->getAccountController();
+        $response          = $accountController->getResponse();
+        /* @todo - create token and add to params */
+        $params = array('token' => $customer->getId());
+        $redirectUrl = Mage::getUrl('shopgate-register/customer_account/create', $params);
+        $response->setRedirect($redirectUrl);
+        $response->sendResponse();
+        exit;
```

Comment:
Magento code should not be exited this way. Maybe try ```Mage::app()->getFrontController()->sendResponse();```?

Reply:
thats also not working.
I can use "$response->sendHeadersAndExit();" but internal this works also with "exit"

Reply:
yeah, that sounds better :/ never seen an exit call in an observer, but I am guessing there is no other way as the redirect probably gets rewritten by Core.